[Sendgrid] - Doc update for new Sendgrid send email Action (#7307)
segmentio/segment-docs: src/connections/destinations/catalog/actions-sendgrid/index.md

{% include components/actions-fields.html %}

> ## Additional details for the 'Send email with Dynamic Template' Action 

Reviewer: Actions are typically represented in Title Case, so I've made that update here.

```suggestion
## Additional details for the Send Email With Dynamic Template Action 
```

### Usage

> The 'Send email with Dynamic Template' Action can be used to send emails via Sendgrid using [Sendgrid Dynamic Templates](https://www.twilio.com/docs/sendgrid/ui/sending-email/how-to-send-an-email-with-dynamic-templates). The Dynamic Template being used must already exist in Sendgrid. Use the Action field named 'Dynamic Template Data' to populate values in the Dynamic Template. 

Reviewer: ```suggestion
The [Send Email With Dynamic Template](#send-email-with-dynamic-template) Action can be used to send emails through SendGrid using [SendGrid Dynamic Templates](https://www.twilio.com/docs/sendgrid/ui/sending-email/how-to-send-an-email-with-dynamic-templates){:target="_blank”}. The Dynamic Template you use must already exist in SendGrid. Use the Action field [Dynamic Template Data](#dynamic-template-data) to populate values in the Dynamic Template. 
```

> ### Contacts
> Emails will be sent to the specified email addresses, even if they are not listed as Contacts in SendGrid Marketing Campaigns.

Reviewer: ```suggestion
### Contacts
SendGrid sends emails to the email addresses you specify, even if they are not listed as Contacts in SendGrid Marketing Campaigns.
```

### Sendgrid API Key

> Segment and SendGrid recommend defining the SendGrid API key within a subuser account and ensuring the domain being used is authenticated under that same subuser account. The 'Send email with Dynamic Template' Action requires the Sendgrid API Key to have the following scopes assigned:   

Reviewer: ```suggestion
Segment and SendGrid recommend that you define the SendGrid API key within a subuser account and the domain is authenticated under that same subuser account. The Send Email With Dynamic Template Action requires that the  SendGrid API Key has the following scopes assigned:   
```

> 1. Category Management: full
> 2. IP Management: full
> 3. IP Management: full
> 4. Template Engine: full

Reviewer: Do these need to be updated in order?

```suggestion
- Category Management: full
- IP Management: full
- IP Management: full
- Template Engine: full
```

> ## Additional details for the 'Upsert Contact' Action 

Reviewer: ```suggestion
## Additional details for the Upsert Contact Action 
```

If you want to view any other custom user traits in the Marketing Campaigns list dashboard, you must create a [Custom Field inside Marketing Campaigns’s UI](https://docs.sendgrid.com/ui/managing-contacts/custom-fields#creating-custom-fields){:target="_blank"} of the traits in your identify calls. Note that you do not need to map all user.traits you are sending inside Marketing Campaigns. You only need to create Custom Fields of the traits you want to see in your list view.

### Custom Fields
To send custom fields/user traits to Marketing Campaigns you need to create the field first in Marketing Campaigns for each trait you want sent to Marketing Campaigns. Then when you call identify with keys that match those traits they will appear in your Marketing Campaigns list.

For any other custom traits just add a Custom Field inside of SendGrid Marketing Campaigns with a tag that matches the key you are using in your identify call.

### Recording userId
To record a Segment userId in SendGrid Marketing Campaigns, you must pass the userID as a trait on your identify() calls. SendGrid does not automatically map the Segment userID to any Marketing Campaigns properties.

> ### Sendgrid API Key

Reviewer: ```suggestion
### SendGrid API Key
```

> The 'Upsert Contact' Action requires the Sendgrid API Key to have the following scopes assigned:

Reviewer: ```suggestion
The Upsert Contact Action requires the SendGrid API Key to have the following scopes:
```

> 1. Marketing: full

Reviewer: ```suggestion
- Marketing: full
```
